### PR TITLE
Adding text on "orange bars of hell"

### PR DESCRIPTION
**Pull request by stuart-presnell**
As discussed in this Zulip thread (https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Troubleshooting.20mathlib), this page is the only relevant result that's returned by a Google search for "troubleshooting mathlib", so it seems worthwhile to put some more general troubleshooting help here.

### CODE REVIEW

File: templates/leanproject.md [hunk 1]
```diff
@@ -253,3 +253,27 @@ If `leanproject` ends with a mysterious error message, you can run it
...
+
+* The first thing to try if this occurs is to close any inactive editor 
+tabs and then restart Lean. (In VSCode, press `ctrl-shift-p` or `cmd-shift-p` 
+to open the Command Palette and then run the command `Lean: Restart`). 
+
+* If the problem persists after restarting, this may be because your local 
+copy of the 
+(pre-compiled olean files)[https://leanprover-community.github.io/glossary.html#cache] 
+is mismatched or faulty in some other way, and so Lean is having to (re)compile 
+all the imported mathlib files.  This taks a long time, making Lean appear
```
Comment (PatrickMassot) [inline templates/leanproject.md:271]:
```suggestion
all the imported mathlib files.  This takes a long time, making Lean appear 
```

File: templates/leanproject.md [hunk 2]
```diff
@@ -253,3 +253,27 @@ If `leanproject` ends with a mysterious error message, you can run it
...
+to open the Command Palette and then run the command `Lean: Restart`). 
+
+* If the problem persists after restarting, this may be because your local 
+copy of the 
+(pre-compiled olean files)[https://leanprover-community.github.io/glossary.html#cache] 
+is mismatched or faulty in some other way, and so Lean is having to (re)compile 
+all the imported mathlib files.  This takes a long time, making Lean appear 
+unresponsive.  In this case, make sure any changes in your current branch 
+are committed, switch to the `master` branch, and then re-download the mathlib 
+cache by typing `leanpkg configure && leanproject get-mathlib-cache` at the terminal.
```
Comment (robertylewis) [inline templates/leanproject.md:274]: I think this is confused about whether you're working on mathlib or a project that depends on mathlib. If it's a dependent project, then switching to master won't do anything. But these problems certainly do show up in external projects, especially if you accidentally touch a file in `_target`.

Comment (stuart-presnell) [reply templates/leanproject.md:274]: Good point, I was only thinking in the context of working on mathlib. Should we have different advice here for the two different cases, or a parenthetical note?

Comment (stuart-presnell) [reply templates/leanproject.md:274]: Or is what I've written just incorrect in either case?

Comment (eric-wieser) [reply templates/leanproject.md:274]: The part about switching to master is bad advice here, as I think I mentioned in a recent zulip thread.

**Comment by semorrison** [comment]
I think this is no longer relevant now that we're switching to Lean 4. Apologies in any case for letting a PR languish for so long!

**Pull request closed without merging**